go-controller/pkg/ovn/pods.go

```diff
@@ -331,9 +330,26 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) error {
 
 	logrus.Debugf("Annotation values: ip=%s ; mac=%s ; gw=%s\nAnnotation=%s",
 		podCIDR, podMac, gatewayIP, marshalledAnnotation)
-	if err = oc.kube.SetAnnotationOnPod(pod, "ovn", marshalledAnnotation); err != nil {
-		return fmt.Errorf("Failed to set annotation on pod %s - %v", pod.Name, err)
+
+	// Need to set pod annotation using both legacy and new key. This is in case if there
+	// are some nodes that have not been upgraded to understand the new Pod annotation
+	err = oc.kube.SetAnnotationOnPod(pod, OvnPodAnnotationName, marshalledAnnotation)
+	if err != nil {
+		logrus.Errorf("Failed to set legacy annotation on pod %s - %v", pod.Name, err)
+	}
+	err = oc.kube.SetAnnotationOnPod(pod, OvnPodAnnotationLegacyName, marshalledAnnotation)
+	if err != nil {
+		logrus.Errorf("Failed to set annotation on pod %s - %v", pod.Name, err)
 	}
 
 	return nil
 }
+
+func getPodNetworkAnnotation(annotation map[string]string) string {
```

Comment:
@girishmg we should just do this internally in UnmarshalPodAnnotation() since that's the only place that uses it. Eg, have UnmarshalPodAnnotation() just take map[string]string and pull the right annotation out.